### PR TITLE
🛡️ Sentry: Add HLC correctness tests

### PR DESCRIPTION
**Pull request by madmax983**
Added robust unit tests to `src/core/hlc.rs` to ensure correctness of the Hybrid Logical Clock implementation, covering standard usage and edge cases like logical counter overflow.

---
*PR created automatically by Jules for task [7325980442448105768](https://jules.google.com/task/7325980442448105768) started by @madmax983*

### CODE REVIEW

File: src/core/hlc.rs [hunk 1]
```diff
@@ -281,3 +281,134 @@ impl From<i64> for HybridTimestamp {
...
+        let ts = valid.unwrap();
+        assert_eq!(ts.wallclock(), 1000);
+        assert_eq!(ts.logical(), 0);
+
+        let invalid = HybridTimestamp::new(MAX_VALID_TIMESTAMP + 1, 0);
+        assert!(invalid.is_err());
+        match invalid {
+            Err(TemporalError::InvalidTimestamp { .. }) => {}
+            _ => panic!("Expected InvalidTimestamp error"),
+        }
```
[the remark this answers is not in the document]
Comment (madmax983) [reply src/core/hlc.rs]: > medium
> 
> This `match` statement can be simplified using the `assert!(matches!(...))` macro for more concise and idiomatic Rust code when asserting error types.
> 
>         assert!(matches!(invalid, Err(TemporalError::InvalidTimestamp { .. })));

@jules
